core/data/src/main/java/blue/starry/tokidokiroppou/core/data/api/EGovLawApiClient.kt

```
@@ -64,7 +72,66 @@ class EGovLawApiClient @Inject constructor(
+        if (trimmedQuery.isBlank()) {
+            return emptyList()
+        }
+
+        return try {
+            Timber.d("Searching e-Gov laws: %s", trimmedQuery)
+            val response = httpClient.get("$BASE_URL/keyword") {
+                parameter("keyword", trimmedQuery)
+            }
+            val body = response.bodyAsText()
+            parseKeywordSearchResponse(body)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)